addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,158 @@
+import requests
+
+from odoo import models, fields, _, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_ro_edi_document_ids = fields.One2many(
+        comodel_name='l10n_ro_edi.document',
+        inverse_name='invoice_id',
+    )
+    l10n_ro_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        compute='_compute_l10n_ro_edi_state',
+        store=True,
+    )
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sending(self, invoice, key_loading: str):
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending',
+            'key_loading': key_loading,
+        })
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sending_failed(self, invoice, message: str):
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending_failed',
+            'message': message,
+        })
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sent(self, invoice, result: dict):
+        document = self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sent',
+            'key_signature': result['key_signature'],
+            'key_certificate': result['key_certificate'],
+        })
+        document.attachment_id = self.env['ir.attachment'].sudo().create({
+            'name': invoice._l10n_ro_edi_get_attachment_file_name(),
+            'raw': result['attachment_raw'],
+            'res_model': self._name,
+            'res_id': document.id,
+            'type': 'binary',
+            'mimetype': 'application/xml',
+        })
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_l10n_ro_edi_state(self):
+        self.l10n_ro_edi_state = False
+        for move in self:
+            for document in move.l10n_ro_edi_document_ids.sorted():
+                if document.state in ('invoice_sending', 'invoice_sent'):
+                    move.l10n_ro_edi_state = document.state
+                    break
+
+    @api.depends('l10n_ro_edi_document_ids', 'l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or an OK response has been received """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            received_error_document = (move.l10n_ro_edi_state == 'invoice_sending' and
+                                       all(doc.state != 'invoice_sending_failed' for doc in move.l10n_ro_edi_document_ids))
+            if move.l10n_ro_edi_state == 'invoice_sent' or received_error_document:
```

Comment:
If you received an error, why can't you just abord the whole thing and reset to draft? :thinking:

Reply:
We can do that, but is that not too intrusive for the user?